### PR TITLE
Switch build system to Dune

### PR DESCRIPTION
**Pull request by dra27**
Working recently on the features which led to #17 and also https://github.com/hannesm/logs-syslog/pull/5, https://github.com/hannesm/logs-syslog/pull/8 and https://github.com/hannesm/logs-syslog/pull/9 I experienced first-hand how suboptimal *developing* with opam pins is for someone contributing across several libraries.

As a retrospect to the work, I offer a switch of build system to Dune - as is so often the case, the diff deletes more than it adds. I've checked that the build is equivalent, at least as far as findlib is concerned. Dune is somewhat opinionated about the names of library archives, so they have to have underscores rather than hyphens, but this should have no effect since the findlib package name is unchanged.

I've written about the experience [here](http://www.dra27.uk/blog/platform/2018/08/15/dune-vendoring.html), and I hope the comparison may provide persuasion, if needed, for why Dune is a good upgrade.

I haven't used it, but please note @samoht's [dune-release](https://github.com/samoht/dune-release) tool, which I believe provides everything which `topkg-care` did before.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 *.swp
-*.native
-*.byte
-*.install
 _build
+syslog-message.install
+src/.merlin
+test/.merlin
```
Comment (Leonidas-from-XIV) [inline .gitignore]: I think it makes sense to just ignore all `.merlin` files

File: dune-project [hunk 2]
```diff
@@ -0,0 +1,3 @@
+(lang dune 1.1)
+(name syslog-message)
+(version 0.0.2)
```
Comment (Leonidas-from-XIV) [inline dune-project]: I think this should be moved to a Changelog file which can be then used by dune-release to create the proper tarballs for releases.

**Review by Leonidas-from-XIV — APPROVED** [review]
I have some small additions but as someone who filed such PRs I am happy someone lese did the work of converting to dune :+1:

File: .gitignore (hunk 1 above)
Comment (dra27) [reply .gitignore]: Sure, I have little opinion on that!

File: dune-project (hunk 2 above)
Comment (dra27) [reply dune-project]: I don't understand what you mean - `dune-project` is a Dune system file, so move what?

**Comment by verbosemode** [comment]
I think we can also get rid of top.ml, since I've only created it for convenient manual testing of the code in the repl. It seems like [Dune](https://jbuilder.readthedocs.io/en/latest/usage.html#launching-the-toplevel-repl) has a builtin for this.

File: dune-project (hunk 2 above)
Comment (Leonidas-from-XIV) [reply dune-project]: Sorry I'm on a train with bad connectivity.

What I meant was to remove the version information from the dune-project file and instead move it to the CHANGELOG.md, which is picked up by dune-release and used for tags, releases and release changelogs by dune-release. This avoids duplicating the version information and forgetting to update it in one place and also simplifies releases and opam-repository PRs.

Comment (dra27) [reply dune-project]: Gotcha - ultimately the plan is that `dune-project` becomes the source of truth for the version and it gets propagated elsewhere, but at the moment `dune-release` does everything that's needed, so I've removed it.

**Comment by dra27** [comment]
@verbosemode - it is indeed supposed to, although it's not working for me at the moment, so I've left `top.ml` for now!

**Comment by verbosemode** [comment]
Nice blog post and thanks again for contributing.

**Pull request merged**